File: DarkOrbit 7.5.3/Net/netty/commands/PetHeroActivationCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class PetHeroActivationCommand
    {
        public const short ID = 31487;

        public static byte[] write(int ownerId, int petId, short shipType, short expansionStage, string petName, short factionId, int clanId, short petLevel, string clanTag, int x, int y, int petSpeed)
        {
            var param1 = new ByteArray(ID);
            param1.writeInt(ownerId);
            param1.writeInt(petId);
            param1.writeShort(shipType);
            param1.writeShort(expansionStage);
            param1.writeUTF(petName);
            param1.writeShort(factionId);
            param1.writeInt(clanId);
            param1.writeShort(petLevel);
            param1.writeUTF(clanTag);
            param1.writeInt(x);
            param1.writeInt(y);
            param1.writeInt(petSpeed);
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/PetHeroActivationCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

/**
Created by LEJYONER on 21/09/2017.
*/

public class PetHeroActivationCommand
        implements ServerCommand {

    public static int ID = 10333;
    public int ownerId, petId, expansionStage, petLevel, x, y, petSpeed, clanId;
    public String petName, clanTag;  
    public class_365 varU1q;
    public short shipType, factionId;
    
    public PetHeroActivationCommand(int param1, int param2, short param3, int param4, String param5, short param6, int param7, 
    		int param8, String param9, int param10, int param11, int param12, class_365 param13) {
        this.ownerId = param1;
        this.petId = param2;
        this.shipType = param3;
        this.expansionStage = param4;
        this.petName = param5;
        this.factionId = param6;
        this.clanId = param7;
        this.petLevel = param8;
        this.clanTag = param9;
        this.x = param10;
        this.y = param11;
        this.petSpeed = param12;
        this.varU1q = param13;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private void writeInternal(DataOutputStream param1) {
        try {
            param1.writeShort(12244);
            param1.writeShort(this.shipType);
            param1.writeInt(this.ownerId >>> 11 | this.ownerId << 21);
            this.varU1q.write(param1);
            param1.writeInt(this.petId << 12 | this.petId >>> 20);
            param1.writeUTF(this.petName);
            param1.writeUTF(this.clanTag);
            param1.writeInt(this.y >>> 15 | this.y << 17);
            param1.writeShort(this.expansionStage);
            param1.writeInt(this.clanId << 5 | this.clanId >>> 27);
            param1.writeInt(this.petSpeed >>> 1 | this.petSpeed << 31);
            param1.writeShort(this.factionId);
            param1.writeInt(this.x << 3 | this.x >>> 29);
            param1.writeShort(this.petLevel);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}